pensieve/pensieve.py

```diff
@@ -245,99 +173,165 @@ def __init__(self, text, par_id=None, doc=None):
-        for thing in things:
-            words_dict['things'][thing.text] += 1
-        return words_dict
-
-    def sanitize_places_and_things(self, freq_cut=100):
+        times = []
+        include_types = ['DATE', 'TIME', 'EVENT']
+        for time in textacy.extract.named_entities(self.spacy_doc,
+                                                   include_types=include_types):
+            times.append(time)
+        return times
+
+    def extract_verbs(self, n_verbs=3):
         """
-        Get a sanitized version of paragraph places and things.
-        Get places that don't appear in the most common doc names, and
-        things that don't appear in the most common doc places.
-
-        Args:
-            freq_cut: frequency to define "commonness" within the doc
+        Extract verbs in the paragraph
+        """
+        verb_ranking = [(x, x.rank) for x in textacy.spacy_utils.get_main_verbs_of_sent(self.spacy_doc)]
+        verbs = []
+        for verb in sorted(verb_ranking, key=lambda x: -x[1])[:n_verbs]:
+            text = verb[0].lemma_
+            if text[0] in string.ascii_lowercase:
+                verbs.append(text)
+        return verbs
+
+    def extract_names(self):
+        """
+        Extract names in paragraph
+        """
+        names = []
+        for name in textacy.extract.named_entities(self.spacy_doc,
+                                                   include_types=['PERSON']):
+            if name.text[-2] not in string.ascii_lowercase:
+                names.append(name.text[:-2])
+            else:
+                names.append(name.text)
+        return names
 
-        Returns:
-            places: sanitized place names
-            things: sanitized thing names
+    def extract_places(self):
+        """
+        Extract places in paragraph
         """
-        doc_names = dict(self.doc.words['names'].most_common(freq_cut))
-        doc_places = dict(self.doc.words['places'].most_common(freq_cut))
         places = []
-        things = []
-        for place in self.words['places']:
-            if place not in doc_names:
-                places.append(place)
-        for thing in self.words['things']:
-            if thing not in doc_places and thing not in places:
-                things.append(thing)
-        return places, things
-
-    def gen_mem_activity(self, verbs_cut=500):
+        include_types = ['LOC', 'GPE', 'FACILITY']
+        for place in textacy.extract.named_entities(self.spacy_doc,
+                                                include_types=include_types):
+            if place.text != 'Hagrid':
+                places.append(place.text)
+        return places
+
+    def extract_things(self):
         """
-        Determine the key activities in a paragraph by removing
-        verbs commonly used in the doc
-
-        Args:
-            verbs_cut: frequency of verb in doc to cut on
+        Extract things mentioned in paragraph
```

Comment:
IF I understand your comment correctly, this is what I had intended by the sanitized palces and things function. My idea was to remove the common names from places, and then remove common names and common places from things